src/components/settings-modal/settings-modal.js

```diff
@@ -80,19 +90,49 @@ const SettingsModal = ({
     hasClickedApplyAndClose,
     showFeatureHintsValue,
     isPrettyNameValue,
+    showDatasetPreviewsValue,
     onToggleFlag,
     onToggleShowFeatureHints,
     onToggleIsPrettyName,
+    onToggleShowDatasetPreviews,
     showSettingsModal,
     toggleFlags,
   ]);
 
+  useEffect(() => {
+    const updatePreferences = async () => {
+      try {
+        const response = await fetch('/api/preferences', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            showDatasetPreviews: showDatasetPreviewsValue,
+          }),
+        });
+
+        if (!response.ok) {
+          throw new Error('Failed to update preferences');
+        }
+
+        const result = await response.json();
+        console.log('Preferences updated successfully:', result);
+      } catch (error) {
+        console.error('Error updating preferences:', error);
+      }
+    };
+
+    updatePreferences();
+  }, [showDatasetPreviewsValue]);
```

Comment:
I see few problems related to api calls
1. On first load every time it make `api/preferences` api call
2. Whenever user clicks on toggle button it make `api/preferences `api call, ideally it should call only when clicked on save button.
3. When user clicks on toggle button and it saves the preferences even after user clicks on cancel button.

Reply:
I moved the API call to update preferences inside the `useEffect` that is triggered when the "Save and apply" button is clicked.